wsmaster/che-core-api-devfile/src/test/java/org/eclipse/che/api/devfile/server/convert/ProjectConverterTest.java

```diff
@@ -131,9 +139,59 @@ public void testCloningIntoProjectsRootFails() throws Exception {
     ProjectImpl devfileProject =
         new ProjectImpl(
             "myProject",
-            new SourceImpl("git", "https://github.com/eclipse/che.git", "master"),
+            new SourceImpl("git", "https://github.com/eclipse/che.git", "master", null, null, null),
             "not/../in/root/../..");
 
     projectConverter.toWorkspaceProject(devfileProject);
   }
+
+  @Test(expectedExceptions = DevfileException.class)
```

Comment:
I think it would be easier to read if `startPoint`, `Tag`, `CommitId` are separated with `or`/`and` -> `testOnlyOneOfStartPointOrTagAllowed`